### PR TITLE
SAMZA-1888: Kafka consumer improvements

### PR DESCRIPTION
**Pull request by sborya**

### CODE REVIEW

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java [hunk 1]
```diff
@@ -16,6 +16,24 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+/*
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: nit: duplicate license header.

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java [hunk 2]
```diff
@@ -630,11 +601,58 @@ public void validateStream(StreamSpec streamSpec) throws StreamValidationExcepti
...
+  }
+
+  private Supplier<ZkUtils> getZkConnection() {
+    String zkConnect =
+        config.get(String.format(KafkaConfig.CONSUMER_CONFIGS_CONFIG_KEY(), systemName, ZOOKEEPER_CONNECT));
+    if (StringUtils.isBlank(zkConnect)) {
+      throw new SamzaException("Missing zookeeper.connect config for admin for system " + systemName);
+    }
+    return () -> {
+      return ZkUtils.apply(zkConnect, 6000, 6000, false);
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: Can be simplified to 
`
() -> ZkUtils.apply(zkConnect, 6000, 6000, false);
`

Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: Do we need `sessionTimeout` and `connectionTimeout` to be configurable? If not, can we extract this to meaningful constants?

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java [hunk 3]
```diff
@@ -630,11 +601,58 @@ public void validateStream(StreamSpec streamSpec) throws StreamValidationExcepti
...
+      }
       KafkaSystemAdminUtilsScala.deleteMessages(adminClient, offsets);
       deleteMessageCalled = true;
     }
   }
 
+  private Properties createAdminClientProperties() {
+    // populate brokerList from either consumer or producer configs
+    Properties props = new Properties();
+    // included SSL settings if needed
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: Do we want to make this method protected instead and have the extenders determine how they want to generate properties?

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java [hunk 4]
```diff
@@ -564,7 +535,7 @@ public boolean clearStream(StreamSpec streamSpec) {
    * @param spec a StreamSpec object
    * @return KafkaStreamSpec object
    */
-  KafkaStreamSpec toKafkaSpec(StreamSpec spec) {
+  public KafkaStreamSpec toKafkaSpec(StreamSpec spec) {
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: why do we need this to be public?

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java [hunk 5]
```diff
@@ -106,12 +108,13 @@ public KafkaSystemConsumer(Consumer<K, V> kafkaConsumer, String systemName, Conf
 
   /**
    * Create internal kafka consumer object, which will be used in the Proxy.
+   * @param <K> key type for the consumer
+   * @param <V> value type for the consumer
    * @param systemName system name for which we create the consumer
    * @param kafkaConsumerConfig config object for Kafka's KafkaConsumer
-   * @return KafkaConsumer object
+   * @return KafkaConsumer newly create kafka consumer object
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java]: nit: s/create/created

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java [hunk 6]
```diff
@@ -630,11 +601,58 @@ public void validateStream(StreamSpec streamSpec) throws StreamValidationExcepti
   @Override
   public void deleteMessages(Map<SystemStreamPartition, String> offsets) {
     if (deleteCommittedMessages) {
+      if (adminClient == null) {
+        adminClient = AdminClient.create(createAdminClientProperties());
+      }
       KafkaSystemAdminUtilsScala.deleteMessages(adminClient, offsets);
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: We need to invoke start on the adminClient after creation prior to deleting messages right?

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java (hunk 2 above)
Comment (sborya) [reply samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: we discussed it before and decided that in this case we don't want any configs.

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java (hunk 3 above)
Comment (sborya) [reply samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: yes. Will change.

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java (hunk 4 above)
Comment (sborya) [reply samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: this method may be used from a different package.

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java (hunk 6 above)
Comment (sborya) [reply samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java]: no, there is not start in adminClient.

**Pull request closed without merging**